velox/docs/develop/debugging/tracing.rst

```diff
@@ -0,0 +1,146 @@
+=======
+Tracing
+=======
+
+
+Background
+----------
+
+The query trace tool helps analyze and debug query performance and correctness
+issues. It helps prevent interference from the test noises in a production
+environment (such as storage, network etc) by allowing replay of a part of the
+query plan and data set in an isolated environment such as a local machine.
+This is much more efficient for query performance debugging as we don't have to
+replay the whole query in a production environment.
+
+How Tracing Tool Works
+----------------------
+
+The tracing process consists of two phases: tracing and replaying.
+
+**Tracing Phase**
+
+- When the query starts, the task records the metadata including query plan fragment,
+  query configuration, and connector properties.
+- During the query running, each traced operator records the input vectors and saves
```

Comment:
running -> execution

Do we store each vector in a separate file? Or do we store all vectors in the same file?